src/main/java/com/tngtech/configbuilder/util/CommandLineHelper.java

```diff
@@ -32,27 +38,44 @@ public CommandLine getCommandLine(Class configClass, String[] args) {
     public Options getOptions(Class configClass) {
         Options options = configBuilderFactory.createInstance(Options.class);
         for (Field field : annotationHelper.getFieldsAnnotatedWith(configClass, CommandLineValue.class)) {
-            if (field.isSynthetic()) {
-                continue;
+            if (!field.isSynthetic()) {
+                options.addOption(getOption(field, configClass));
             }
-            options.addOption(getOption(field));
         }
         return options;
     }
 
-    @SuppressWarnings("AccessStaticViaInstance")
-    private Option getOption(Field field) {
+    private Option getOption(Field field, Class configClass) {
         CommandLineValue commandLineValue = field.getAnnotation(CommandLineValue.class);
         log.debug("adding command line option {} for field {}", commandLineValue.shortOpt(), field.getName());
         return Option.builder(commandLineValue.shortOpt())
                 .longOpt(commandLineValue.longOpt())
                 .hasArg()
                 .required(commandLineValue.required())
-                .desc(commandLineValue.description())
+                .desc(extractDescriptionString(commandLineValue, configClass))
                 .hasArg(commandLineValue.hasArg())
                 .build();
     }
 
+    private String extractDescriptionString(CommandLineValue commandLineValue, Class configClass) {
+        if (!commandLineValue.description().isEmpty()) {
+            return commandLineValue.description();
+        }
+
+        Set<Method> descriptorMethods = annotationHelper.getMethodsAnnotatedWith(configClass, CommandLineValueDescriptor.class);
+        if (descriptorMethods.isEmpty()) {
+            return "";
+        }
+
+        try {
+            Method descriptorMethod = getOnlyElement(descriptorMethods);
+            descriptorMethod.setAccessible(true);
+            return descriptorMethod.invoke(null, commandLineValue.longOpt()).toString();
+        } catch (Exception e) {
+            throw new ConfigBuilderException(errorMessageSetup.getErrorMessage(InvalidDescriptionMethodException.class), e);
```

Comment:
Could you actually get this error message? I've tried with the following test:
```java
    private static class TestConfigWithInvalidCommandLineValueDescriptor {
        @CommandLineValue(shortOpt = "u", longOpt = "user")
        public String user;
        @CommandLineValueDescriptor
        private static void description1() {
        }
        @CommandLineValueDescriptor
        private static void description2() {
        }
    }

    @Test
    public void testInvalidCommandLineValueDescriptor() {
        assertThatThrownBy(() -> ConfigBuilder.on(TestConfigWithInvalidCommandLineValueDescriptor.class).build())
                .isInstanceOf(ConfigBuilderException.class)
                .hasMessageContaining("invalid or multiple use of the @CommandLineValueDescriptor annotation");
    }
```
which gives me
```
java.lang.NullPointerException
	at com.tngtech.configbuilder.configuration.ErrorMessageSetup.getErrorMessage(ErrorMessageSetup.java:47)
	at com.tngtech.configbuilder.util.CommandLineHelper.extractDescriptionString(CommandLineHelper.java:75)
	at com.tngtech.configbuilder.util.CommandLineHelper.getOption(CommandLineHelper.java:55)
	at com.tngtech.configbuilder.util.CommandLineHelper.getOptions(CommandLineHelper.java:42)
	at com.tngtech.configbuilder.ConfigBuilder.<init>(ConfigBuilder.java:81)
	at com.tngtech.configbuilder.ConfigBuilder.<init>(ConfigBuilder.java:88)
	at com.tngtech.configbuilder.ConfigBuilder.on(ConfigBuilder.java:296)
	at com.tngtech.configbuilder.util.CommandLineHelperTest.testInvalidCommandLineValueDescriptor
```
It seems that `ErrorMessageSetup#initialize` is only called from `ConfigBuilder#build`, whereas `CommandLineHelper.getOptions` is already called from the `ConfigBuilder` constructor.

Unfortunately, the initialization of all the helper classes seems to be quite confusing. ☹

Reply:
I wonder whether we should just add another 
```
initializeErrorMessageSetup(propertyLoader);
```
before the `ConfigBuilder` constructor's
```
        commandLineOptions = commandLineHelper.getOptions(configClass);
```
(We probably cannot actually move the `initializeErrorMessageSetup(propertyLoader)` call from the `build` method, as `propertyLoader` could be configured differently at the the end... so that should probably be conserved.)

Reply:
Hm; I found it more self-consistent to drop the `commandLineOptions` field and evaluate it lazily instead, as it's anyways only needed for printing the help text. That way, also any user-defined configuration of the `propertyLoader` is effective whenever an exception is thrown.